sql/7_0_1-to-7_0_2_upgrade.sql

```diff
@@ -216,3 +216,15 @@ ALTER TABLE `facility` ADD COLUMN `inactive` tinyint(1) NOT NULL DEFAULT '0';
 INSERT INTO `list_options` (`list_id`, `option_id`, `title`, `seq`, `is_default`, `option_value`, `mapping`, `notes`, `codes`, `toggle_setting_1`, `toggle_setting_2`, `activity`) VALUES ('Document_Template_Categories','notification_template','Notification Template',105,0,0,'','','',0,0,1);
 #EndIf
 
+#IfNotRow2D list_options list_id lists option_id default_open_tabs
+-- Create new list Default Open Tabs
+INSERT INTO `list_options` (`list_id`, `option_id`, `title`) VALUES ('lists', 'default_open_tabs', 'Default Open Tabs');
+-- Populate list with sensible defaults based on previous optiosn in globals, assume no tabs are actually active
+INSERT INTO `list_options` (`list_id`, `notes`, `title`, `seq`, `option_id`, `activity`) VALUES ('default_open_tabs', 'main_info.php', 'Calendar', 1, 'cal', '0');
+INSERT INTO `list_options` (`list_id`, `notes`, `title`, `seq`, `option_id`, `activity`) VALUES ('default_open_tabs', '../new/new.php', 'Patient Search / Add', 2, 'pat', '0');
+INSERT INTO `list_options` (`list_id`, `notes`, `title`, `seq`, `option_id`, `activity`) VALUES ('default_open_tabs', '../../interface/main/finder/dynamic_finder.php', 'Patient Finder', 3, 'fin', '0');
```

Comment:
this opens door to directory traversal vulnerability. not sure how to mitigate this. gotta think about it for a bit (likely involves ensuring the flattened path with realpath() function does not lie outside the `$GLOBALS['fileroot']` path).

Reply:
Yep, good call. I'll work on solving this as well.

Reply:
it's these that need to be added to `sql/database.sql`

Reply:
Added to database.sql as well. Address the vulnerability by concatenating the file root to the URL and then running it through `realpath()` - if that returns false that row in the array is removed and we move on; otherwise, the path does exist (within `$fileroot` so we can move forward.

I tested this by creating a new entry in the list called "Test" with a URL set to "/var/log/apache2/error.log" which doesn't survive the load.

For better control, when inserting the row in the list, the URL should assume it starts at the file path (no need to to `../../` for any reason).